clang-tools-extra/clang-tidy/modernize/ReturnBracedInitListCheck.cpp

```diff
@@ -17,11 +17,21 @@ using namespace clang::ast_matchers;
 namespace clang::tidy::modernize {
 
 void ReturnBracedInitListCheck::registerMatchers(MatchFinder *Finder) {
-  // Skip list initialization and constructors with an initializer list.
+  auto SemanticallyDifferentContainer = allOf(
+      argumentCountIs(2), hasArgument(0, hasType(isInteger())),
+      hasType(cxxRecordDecl(hasAnyName("::std::vector", "::std::deque",
```

Comment:
Probably you could include also std::string and base_string to match this constructor:
`basic_string( size_type count, CharT ch,  const Allocator& alloc = Allocator() );`